subsys/bluetooth/crypto_toolbox/include/crypto_toolbox.h

```diff
@@ -0,0 +1,115 @@
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * @brief Cypher based Message Authentication Code (CMAC) with AES 128 bit
+ *
+ * Defined in Core Vol. 3, part H 2.2.5.
+ *
+ * @param key 128-bit key
+ * @param in message to be authenticated
+ * @param len length of the message in octets
+ * @param[out] out message authentication code
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
+ */
+int bt_crypto_toolbox_aes_cmac(const uint8_t *key, const uint8_t *in, size_t len, uint8_t *out);
+
+/**
+ * @brief Cryptographic Toolbox f4
+ *
+ * Defined in Core Vol. 3, part H 2.2.6.
+ *
+ * @param u 256-bit
+ * @param v 256-bit
+ * @param x 128-bit key
+ * @param z 8-bit
+ * @param[out] res
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
+ */
+int bt_crypto_toolbox_f4(const uint8_t *u, const uint8_t *v, const uint8_t *x, uint8_t z,
+			 uint8_t res[16]);
+
+/**
+ * @brief Cryptographic Toolbox f5
+ *
+ * Defined in Core Vol. 3, part H 2.2.7.
+ *
+ * @param w 256-bit
+ * @param n1 128-bit
+ * @param n2 128-bit
+ * @param a1 56-bit
+ * @param a2 56-bit
+ * @param[out] mackey most significant 128-bit of the result
+ * @param[out] ltk least significant 128-bit of the result
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
+ */
+int bt_crypto_toolbox_f5(const uint8_t *w, const uint8_t *n1, const uint8_t *n2,
+			 const bt_addr_le_t *a1, const bt_addr_le_t *a2, uint8_t *mackey,
+			 uint8_t *ltk);
+
+/**
+ * @brief Cryptographic Toolbox f6
+ *
+ * Defined in Core Vol. 3, part H 2.2.8.
+ *
+ * @param w 128-bit
+ * @param n1 128-bit
+ * @param n2 128-bit
+ * @param r 128-bit
+ * @param iocap 24-bit
+ * @param a1 56-bit
+ * @param a2 56-bit
+ * @param[out] check
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
+ */
+int bt_crypto_toolbox_f6(const uint8_t *w, const uint8_t *n1, const uint8_t *n2, const uint8_t *r,
+			 const uint8_t *iocap, const bt_addr_le_t *a1, const bt_addr_le_t *a2,
+			 uint8_t *check);
+
+/**
+ * @brief Cryptographic Toolbox g2
+
+ * Defined in Core Vol. 3, part H 2.2.9.
+ *
+ * @param u 256-bit
+ * @param v 256-bit
+ * @param x 128-bit
+ * @param y 128-bit
+ * @param[out] passkey
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
+ */
+int bt_crypto_toolbox_g2(const uint8_t u[32], const uint8_t v[32], const uint8_t x[16],
+			 const uint8_t y[16], uint32_t *passkey);
+
+/**
+ * @brief Cryptographic Toolbox h6
+ *
+ * Link key conversion defined in Core Vol. 3, part H 2.2.10.
+ *
+ * @param w 128-bit key
+ * @param key_id 32-bit
+ * @param[out] res 128-bit
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval -EIO
```

Comment:
can you add a reason for that error code? e.g. `@retval -EIO Computation failed.`